### PR TITLE
Feat/migrate content folder structure

### PR DESCRIPTION
**Pull request by jmoguilevsky**
## Description

Please include a summary of the change and which issue is fixed/closed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

Closes # (issue)

## Changes

Specify the behavior changes introduced by this pull request

i.e.

- Removed method X from class Y
- Return 400 instead of 500 in Z case
- Refactored class X into a new module

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change). If there's an API change, link the pull request in the [API spec repository](https://github.com/decentraland/catalyst-api-specs) and the accepted [DAO governance poll](https://governance.decentraland.org/)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/decentraland/catalyst/blob/main/docs/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: content/src/service.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ export async function main(program: Lifecycle.EntryPointParameters<AppComponents
   // TODO: move this purgeUploadsDirectory method to a standalone function inside src/logic/ folder
   await components.server.purgeUploadsDirectory()
 
+  await components.contentFolderMigrationManager.run()
```
Comment (menduz) [inline content/src/service.ts:16]: This whole thing "contentFolderMigrationManager" should be a function receiving `components`, not a class

File: content/test/unit/storage/ContentStorageWithStreams.spec.ts [hunk 2]
```diff
@@ -19,15 +19,15 @@ describe('ContentStorage', () => {
   })
 
   it(`When content is stored, then it can be retrieved`, async () => {
-    await storage.storeStream(id, bufferToStream(content))
+    await storage.store(id, bufferToStream(content))
```
Comment (menduz) [inline content/test/unit/storage/ContentStorageWithStreams.spec.ts:22]: Prefer specific functions over overloads

File: yarn.lock [hunk 3]
```diff
@@ -3259,6 +3261,11 @@ cross-spawn@^7.0.0, cross-spawn@^7.0.2, cross-spawn@^7.0.3:
     shebang-command "^2.0.0"
     which "^2.0.1"
 
+crypto@^1.0.1:
```
Comment (menduz) [inline yarn.lock:3264]: Please remove, `crypto` is part of Nodejs std lib

File: content/src/storage/FileSystemContentStorage.ts [hunk 4]
```diff
@@ -76,19 +72,87 @@ export class FileSystemContentStorage implements ContentStorage {
...
+  }
+
+  async fixContentItem(id: string, oldContentPath: string): Promise<void> {
+    return fs.promises.rename(path.join(oldContentPath, id), this.getFilePath(id))
+  }
+
+  create(id: string): UnsavedContentItem {
+    // Check if it doesn't exist
+    return createUnsavedContentItem(id, this.getTmpFilePath(id), this.getFilePath(id))
+  }
```
Comment (menduz) [inline content/src/storage/FileSystemContentStorage.ts:94]: I'd move this to a helper outside the storage component. How will this work for S3 or IPFS? I think tmp file management should be separated from the storage. And adding a storage.storeFromFile(tmpPath) method to handle the copy/upload/renaming

File: yarn.lock (hunk 3 above)
Comment (pedrotambo) [reply yarn.lock:3264]: I think it's part of node v17, but where are on v16?

File: content/src/migrations/ContentFolderMigrationManager.ts [hunk 5]
```diff
@@ -45,7 +45,7 @@ export class ContentFolderMigrationManager {
       pending.push(
         this.queue.add(async () => {
           try {
-            await this.storage.fixContentItem(file.name, this.contentsFolder)
+            await this.storage.storeExistingContentItem(file.name, this.contentsFolder)
```
Comment (menduz) [inline content/src/migrations/ContentFolderMigrationManager.ts:48]: Watchout, the signature may be wrong here

File: content/src/storage/FileSystemContentStorage.ts [hunk 6]
```diff
@@ -84,7 +84,7 @@ export class FileSystemContentStorage implements ContentStorage {
     return (await this.stats(id))?.size
   }
 
-  async fixContentItem(id: string, oldContentPath: string): Promise<void> {
+  async storeExistingContentItem(id: string, oldContentPath: string): Promise<void> {
```
Comment (menduz) [inline content/src/storage/FileSystemContentStorage.ts:87]: Please be more specific
```suggestion
  async storeExistingContentItem(fileInDisk: string, contentHash: string): Promise<void> {
```

**Comment by jmoguilevsky** [comment]
Replaced by #888

**Pull request closed without merging**